client/src/test/java/io/getstream/chat/android/client/parser/adapters/CustomObjectAdapterTest.kt

```diff
@@ -0,0 +1,74 @@
+package io.getstream.chat.android.client.parser.adapters
+
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.parser.ChatParserImpl
+import org.amshove.kluent.shouldBeEqualTo
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class CustomObjectAdapterTest {
+    private val parser = ChatParserImpl()
+
+    @BeforeEach
+    fun `check prerequisites`() {
+        val jsonObject = JsonParser.parseString(json).asJsonObject
+        checkJsonStructureAndValues(jsonObject)
+    }
```

Comment:
Couldn't this be a single test itself, which fails on its own if the check in it fails, instead of failing all tests here?